developer/coding-guide.md

```diff
@@ -0,0 +1,83 @@
+# Coding style guide
+
+A guide to working on code at Reaction, in bite-size steps!
+
+## Git: Issues, branches, commits
+
+Git version control is how we work together as a team. Naming branches and writing commit messages helps us keep a easy-to-understand history of the changes in the project.
+
+Follow these guidelines to make releases and code reviews smoother.
+
+1. Find a GitHub issue for what you're working on.
+
+2. Branch name: Create a branch name, with the ID number of the GitHub issue, in the following style:
+`yourhandle-type-###-slug`
+
+```
+spencern-feat-222-add-email
+machikoyasuda-docs-123-email
+impactmass-fix-341-email-retries
+```
+
+3. Make atomic commits in Angular.js style, with a `(scope)` and `subject`, in the following style:
+
+The subject line: Has a (type) and short description.
+```
+(feat): add sendEmail() job
+(docs): add doc for sendEmailJob() method
+(fix): bug fix for issue
+```
+
+The footer: Notes the issue number:
+```
+Closes #222
+```
+
+All of the types:
+- feat: A new feature
+- fix: A bug fix
+- docs: Documentation only changes
+- style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+- refactor: A code change that neither fixes a bug nor adds a feature
+- perf: A code change that improves performance
+- test: Adding missing or correcting existing tests
+- chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
+
+Note the following styles:
+- use the imperative, present tense: "change" not "changed" nor "changes"
+- don't capitalize first letter
+- no dot (.) at the end
+- use `BREAKING CHANGES:` to note breaking changes
+
+More details: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commit-message-format
+Examples: https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#heading=h.8sw072iehlhg
+
+4. When you're ready to make a pull request, make sure the PR title and body contain the ### of the issue.
```

Comment:
I think all PRs should have the following: 
1-A) ### of the issue they resolve
OR
1-B) Description of the issue they resolve, if unreported or fixing a problem that does not currently have an issue.

2) List of all changes that the PR makes to the functionality of the app, bugs it fixes and how it accomplishes those things if it's not immediately obvious.

3) Steps to test the PR. If it involves more setup than `reaction reset && reaction` the setup steps should be listed, otherwise the steps should be assumed to involve starting from a fresh instance.

Anything else?

Reply:
I think even when there is an issue, a description of how the PR is solving the problem is really helpful. What was wrong in the code and how does this PR correct it.

Maybe something about when it's okay to refactor/fix code and when it is not. Meaning, in bug-fix code I would prefer that the PR **only** contain code that fixes the bug and absolutely nothing else. Is this always true? Seems like the "keep the PR focused" dictum would tend to say that but I see a lot of "replace this lodash method with native" type comments on code that's not new/touched. I am all for improving code when you see an opportunity to do so but I don't know where we are drawing that line.

Reply:
I'd agree with that, and I think it's a hard line to draw in terms of when something needs to exist in a separate PR rather than being included in a PR that has a focus on different things.

Reply:
I guess where I would (try to) draw the line is if the affected code will be tested in the course of testing the PR. As long as we can verify the changed code is not broken during that process than I am okay with it

Reply:
I like that distinction. With that in mind, we should require that test steps should indicate where you can verify the "side-fixes" are also resolved.